Explorer/Assets/Scripts/ECS/Unity/PrimitiveColliders/Systems/InstantiatePrimitiveColliderSystem.cs

```diff
@@ -0,0 +1,135 @@
+
+        protected override void Update(float t)
+        {
+            InstantiateNonExistingColliderQuery(World);
+            TrySetupExistingColliderQuery(World);
+        }
+
+        [Query]
+        [All(typeof(PBMeshCollider), typeof(SDKTransform), typeof(Transform))]
+        [None(typeof(PrimitiveColliderComponent))]
+        private void InstantiateNonExistingCollider(in Entity entity, ref PBMeshCollider sdkComponent, ref Transform transform)
+        {
+            var component = new PrimitiveColliderComponent();
+            Instantiate(setupColliderCases[sdkComponent.MeshCase], ref component, ref sdkComponent, ref transform);
+            World.Add(entity, component);
+        }
+
+        [Query]
+        [All(typeof(PBMeshCollider), typeof(SDKTransform), typeof(Transform), typeof(PrimitiveColliderComponent))]
+        private void TrySetupExistingCollider(
+            ref PrimitiveColliderComponent primitiveColliderComponent,
+            ref PBMeshCollider sdkComponent,
+            ref Transform transform)
+        {
+            if (!sdkComponent.IsDirty) return;
+
+            ISetupCollider setupCollider = setupColliderCases[sdkComponent.MeshCase];
+
+            // Prevent calling an overloaded comparison from UnityEngine.Object
+            if (ReferenceEquals(primitiveColliderComponent.Collider, null))
+                Instantiate(setupCollider, ref primitiveColliderComponent, ref sdkComponent, ref transform);
+            else
+
+                // Just a change of parameters
+                SetupCollider(setupCollider, primitiveColliderComponent.Collider, sdkComponent);
+
+            sdkComponent.IsDirty = false;
+        }
+
+        private void SetupCollider(ISetupCollider setupCollider, Collider collider, in PBMeshCollider sdkComponent)
+        {
+            // Setup collider only if it's gonna be enabled, otherwise there is no reason to [re]generate a shape
+            if (SetColliderLayer(collider, sdkComponent))
+                setupCollider.Execute(collider, sdkComponent);
+        }
+
+        private bool SetColliderLayer(Collider collider, in PBMeshCollider sdkComponent)
+        {
+            uint colliderLayer = sdkComponent.GetColliderLayer();
+
+            GameObject colliderGameObject = collider.gameObject;
+
+            var enabled = true;
+
+            if ((colliderLayer & LAYER_PHYSICS_POINTER) == LAYER_PHYSICS_POINTER)
+                colliderGameObject.layer = PhysicsLayers.DEFAULT_LAYER;
+            else if ((colliderLayer & LAYER_PHYSICS) == LAYER_PHYSICS)
+                colliderGameObject.layer = PhysicsLayers.CHARACTER_ONLY_LAYER;
+            else if ((colliderLayer & LAYER_POINTER) == LAYER_POINTER)
+                colliderGameObject.layer = PhysicsLayers.ON_POINTER_EVENT_LAYER;
+            else if (PhysicsLayers.LayerMaskHasAnySDKCustomLayer(colliderLayer))
+                colliderGameObject.layer = PhysicsLayers.SDK_CUSTOM_LAYER;
+            else if (collider != null)
```

Comment:
Over here collider will always have a value (since it was confirmed in line 71). Look to me that this could be changed just to an else